builtin/providers/aws/data_source_availability_zones.go

```diff
@@ -21,32 +21,65 @@ func dataSourceAwsAvailabilityZones() *schema.Resource {
 				Computed: true,
 				Elem:     &schema.Schema{Type: schema.TypeString},
 			},
+			"state": &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "available",
```

Comment:
I think setting this default would make it impossible to retrieve an unfiltered list, and is a backward-compatibility break.

Could we instead default to no filter, and have the user explicitly set `state = "available"` to get the filtered list? This does make the user do a little more work, but has the advantage of being consistent with how some of our other data sources are working (each new attribute acts as an additional filter constraint) and makes things a little more explicit.